examples/source_separation/README.md

```diff
@@ -0,0 +1,168 @@
+  If the training is carried out on a single node, then the default `"env://"` should do.
+  If the training processes span across multiple nodes, then you need to provide a protocol that
+  can communicate over the network. If you know where the master node is located, you can use
+  `"env://"` in combination with `MASTER_ADDR` and `MASER_PORT` environment variables. If you do
+  not know where the master node is located beforehand, you can use `"file://..."` protocol to
+  indicate where the file to which all the worker process have access is located. For other
+  protocols, please refer to the official documentation.
+
+### Distributed Training Notes
+
+<details><summary>Quick overview on DDP (distributed data parallel)</summary>
+
+DDP is single-program multiple-data training paradigm.
+With DDP, the model is replicated on every process,
+and every model replica will be fed with a different set of input data samples.
+
+- **Process**: Worker process (as in Linux process). There are `P` processes per a Node.
+- **Node**: A machine. There are `N` machines, each of which holds `P` processes.
+- **World**: network of nodes, composed of `N` nodes and `N * P` processes.
+- **Rank**: Grobal process ID (unique across nodes) `[0, N * P)`
+- **Local Rank**: Local process ID (unique only within a node) `[0, P)`
+
+```
+          Node 0                    Node 1                          Node N-1
+┌────────────────────────┐┌─────────────────────────┐     ┌───────────────────────────┐
+│╔══════════╗ ┌─────────┐││┌───────────┐ ┌─────────┐│     │┌─────────────┐ ┌─────────┐│
+│║ Process  ╟─┤ GPU: 0  ││││ Process   ├─┤ GPU: 0  ││     ││ Process     ├─┤ GPU: 0  ││
+│║ Rank: 0  ║ └─────────┘│││ Rank:P    │ └─────────┘│     ││ Rank:NP-P   │ └─────────┘│
+│╚══════════╝            ││└───────────┘            │     │└─────────────┘            │
+│┌──────────┐ ┌─────────┐││┌───────────┐ ┌─────────┐│     │┌─────────────┐ ┌─────────┐│
+││ Process  ├─┤ GPU: 1  ││││ Process   ├─┤ GPU: 1  ││     ││ Process     ├─┤ GPU: 1  ││
+││ Rank: 1  │ └─────────┘│││ Rank:P+1  │ └─────────┘│     ││ Rank:NP-P+1 │ └─────────┘│
+│└──────────┘            ││└───────────┘            │ ... │└─────────────┘            │
+│                        ││                         │     │                           │
+│ ...                    ││ ...                     │     │ ...                       │
+│                        ││                         │     │                           │
+│┌──────────┐ ┌─────────┐││┌───────────┐ ┌─────────┐│     │┌─────────────┐ ┌─────────┐│
+││ Process  ├─┤ GPU:P-1 ││││ Process   ├─┤ GPU:P-1 ││     ││ Process     ├─┤ GPU:P-1 ││
+││ Rank:P-1 │ └─────────┘│││ Rank:2P-1 │ └─────────┘│     ││ Rank:NP-1   │ └─────────┘│
+│└──────────┘            ││└───────────┘            │     │└─────────────┘            │
+└────────────────────────┘└─────────────────────────┘     └───────────────────────────┘
+```
+
+</details>
+
+### SLURM
```

Comment:
submitit is great for user experience: https://github.com/facebookresearch/detr/blob/master/run_with_submitit.py

Reply:
E.g. can do something like
```
python run_with_submitit.py --nodes 4 --timeout 3000 --coco_path /datasets01/COCO/022719
```

Reply:
See https://github.com/facebookincubator/submitit for submitit repo

Reply:
I looked at it, but it feels like it adds the more learning curve to go through this example.
The code is right now completely agnostic to SLURM yet runnable with SLURM.
I think that leaving the decision of the adaptation of submitit to users will keep our code simpler.
As long as the bare minimum of how to use in SLURM environment is explained in README, then it should do.